db/src/tx.rs

```diff
@@ -633,13 +654,16 @@ impl<'conn, 'a> Tx<'conn, 'a> {
             }
         }
 
-        // Transact [:db/add :db/txInstant NOW :db/tx].
-        // TODO: allow this to be present in the transaction data.
-        non_fts_one.push((self.tx_id,
-                          entids::DB_TX_INSTANT,
-                          self.schema.require_attribute_for_entid(entids::DB_TX_INSTANT).unwrap(),
-                          TypedValue::Instant(self.tx_instant),
-                          true));
+        tx_instant = self.tx_instant.unwrap_or(::now());
+
+        // Transact [:db/add :db/txInstant NOW :db/tx] if it doesn't exist.
+        if self.tx_instant == None {
```

Comment:
You always set `tx_instant` right above this. I don't think you need this conditional check (or the comment).

Reply:
Yeah, but the point is: we've filtered out all txInstant-setting datoms from the input. Every tx is supposed to have one. So `tx_instant` (the local var) is the value we want, and we should always add it to the output. Right?

Reply:
Oh, you stripped out the `continue`. Got it.